docs/README.md

```diff
@@ -76,26 +71,15 @@ you should expect an object that looks like that as the return value of `useProd
 
 ℹ️ To have the full type definition in your development environment, be sure to run `vtex setup` in your project to install all TypeScript types exported by this app.
 
-### `ProductContextProvider`
-
-The `ProductContextProvider` component is capable of providing the `ProductContext` to all of it's children. 
-
-It expects to receive the following props:
-
-```ts
-interface ProductAndQuery {
-  query: Record<string, any>
-  product: MaybeProduct
-}
-```
+### `ProductContext`
 
-and will also wrap it's children with the `ProductDispatchContext` component described below.
+The `ProductContext` entry point exports the React context in it's raw form. This is only useful if you want to reference it directly.
 
-### `ProductDispatchContext`
+Be aware that most of the times, what you're looking for is the `useProduct` hook described below.
 
-The `ProductDispatchContext` entry point exports a React context in it's raw form. This is only useful if you want to reference it directly. Most likely you don't need to use this, since the `ProductContextProvider` already wraps it's children with a provider for this context.
+### `useProductDispatch`
 
-This is the context that exposes functions you can use to manipulate the nearest `ProductContext`. It's value is a `dispatch` function, capable of performing the following `actions`:
+This hooks returns a `dispatch` function you can use to manipulate the nearest `ProductContext`. This function is capable of performing the following `actions`:
 
 ```ts
 type Actions =
```

Comment:
This is not legible

Reply:
What do you mean @klzns ?

Reply:
I tried to move each one to its own type, but it got way worse: 

```ts
type ACTION_SELECT_IMAGE_VARIATION = Action<
  'SELECT_IMAGE_VARIATION',
  { args: { selectedImageVariationSKU: string | null } }
>;
type ACTION_SET_QUANTITY = Action<
  'SET_QUANTITY',
  { args: { quantity: number } }
>;
type ACTION_SKU_SELECTOR_SET_VARIATIONS_SELECTED = Action<
  'SKU_SELECTOR_SET_VARIATIONS_SELECTED',
  { args: { allSelected: boolean } }
>;
type ACTION_SET_BUY_BUTTON_CLICKED = Action<
  'SET_BUY_BUTTON_CLICKED',
  { args: { clicked: boolean } }
>;
type ACTION_SKU_SELECTOR_SET_IS_VISIBLE = Action<
  'SKU_SELECTOR_SET_IS_VISIBLE',
  { args: { isVisible: boolean } }
>;
type ACTION_SET_SELECTED_ITEM = Action<
  'SET_SELECTED_ITEM',
  { args: { item: Item | undefined | null } }
>;
type ACTION_SET_ASSEMBLY_OPTIONS = Action<
  'SET_ASSEMBLY_OPTIONS',
  {
    args: {
      groupId: string;
      groupItems: AssemblyOptionItem[];
      groupInputValues: InputValues;
      isValid: boolean;
    };
  }
>;
type ACTION_SET_PRODUCT = Action<
  'SET_PRODUCT',
  { args: { product: MaybeProduct } }
>;

type Actions =
  | ACTION_SELECT_IMAGE_VARIATION
  | ACTION_SET_QUANTITY
  | ACTION_SKU_SELECTOR_SET_VARIATIONS_SELECTED
  | ACTION_SET_BUY_BUTTON_CLICKED
  | ACTION_SKU_SELECTOR_SET_IS_VISIBLE
  | ACTION_SET_SELECTED_ITEM
  | ACTION_SET_ASSEMBLY_OPTIONS
  | ACTION_SET_PRODUCT;
```

Reply:
Maybe you could just specify the action names?

Reply:
I suggest to document as markdown text